repository_pool.go

```diff
@@ -175,100 +174,33 @@ func (p *RepositoryPool) Add(repo repository) error {
 
 // AddGit checks if a git repository can be opened and adds it to the pool. It
 // also sets its path as ID.
-func (p *RepositoryPool) AddGit(path string) (string, error) {
+func (p *RepositoryPool) AddGit(path string) error {
 	return p.AddGitWithID(path, path)
 }
 
 // AddGitWithID checks if a git repository can be opened and adds it to the
 // pool. ID should be specified.
-func (p *RepositoryPool) AddGitWithID(id, path string) (string, error) {
+func (p *RepositoryPool) AddGitWithID(id, path string) error {
 	_, err := git.PlainOpen(path)
 	if err != nil {
-		return "", errRepoCannotOpen.Wrap(err, path)
+		return errRepoCannotOpen.Wrap(err, path)
 	}
 
-	err = p.Add(gitRepo(id, path))
-	if err != nil {
-		return "", err
-	}
-
-	return path, nil
+	return p.Add(gitRepo(id, path))
 }
 
-// AddDir adds all direct subdirectories from path as git repos. Prefix is the
-// number of directories to strip from the ID.
-func (p *RepositoryPool) AddDir(prefix int, path string) error {
-	dirs, err := ioutil.ReadDir(path)
-	if err != nil {
-		return err
-	}
-
-	for _, f := range dirs {
-		if f.IsDir() {
-			pa := filepath.Join(path, f.Name())
-			id := IDFromPath(prefix, pa)
-			if _, err := p.AddGitWithID(id, pa); err != nil {
-				logrus.WithFields(logrus.Fields{
-					"id":    id,
-					"path":  pa,
-					"error": err,
-				}).Error("repository could not be added")
-			} else {
-				logrus.WithField("path", pa).Debug("repository added")
-			}
-		}
+// AddSivaFile adds to the pool the given file if it's a siva repository,
+// that is, has the .siva extension
+func (p *RepositoryPool) AddSivaFile(path string) error {
+	file := filepath.Base(path)
+	if !strings.HasSuffix(file, ".siva") {
```

Comment:
I would separate the logic related to check if the path is from a siva file, and add a siva repository to the repository pool. Per example, as we have a `isGitRepo` function, we should have a `isSivaFile` function to check if we can add that file to the pool.